### PR TITLE
Front-End tests with Cypress

### LINKED ISSUE #2358: Bugs with Media field type: Trying to add new content entry says 'no fields exist', and Media fields are not showing in relationship data.
**Informations**
- **Node.js version**: 10.11.0
- **npm version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.14.5
- **Database**: mysql 
- **Operating system**: OSX High Sierra

**What is the current behavior?**
If I add a new content type that only defined fields that are 'Media' fields, then save it, then go to try and add content, it says that no fields exist for this content type. If I then add any other field type, I can add a new content entry. 
It also does not show these Media fields as selectable 'Attribute fields' in the Content Manager type editor.
Furthermore, if I define a relationship between an object and this other type (with Media fields), then when I query the api for the first object data, it returns the relationship data, but does not include the Media fields. This is a larger bug.

As an elaboration/example:
If I make a content type 'product', with Media fields, then query the /products api, I see the Media fields. If I then have a content type such as 'productgroup', that defines a relationship to 'products', then when I query the /productgroups api, it returns the related data, but omits the Media fields (seems to not respect the upload_file_morph table?)

**Steps to reproduce the problem**
Define a new Content Type with only 'Media' fields. Then try to add a new piece of content for it.
Define a new relationship to a content type with media fields, and ask for the former, and you will not see the Media fields in the relationship data.

**What is the expected behavior?**
It should allow new content to be added using types that only define Media fields. It should also return the Media fields in the relationship data.

**Suggested solutions**
For now I'll just add a dummy field, but perhaps the Media field is new and not being recognized by the content entry validation system. As far as the relationship data goes, I will have to query the api after the first data is returned (ie. manually retrieve the relationships).

### PR DESCRIPTION
**Pull request by soupette**
My PR is a:
- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix #2358 
- [x] 💅 Enhancement
- [x] 🚀 New feature

Main update on the:
- [ ] Admin
- [ ] Documentation
- [x] Framework
- [x] Plugin

Add end-to-end test with Cypress (cypress.io). This PR will only test the front-end part of Strapi.

**TODO**
- **Content Type Builder**
  - [x] Test build
  - [x] Test create API
  - [x] Test update API and check that the `plugin_content-manager_schema` document is ok
  - [x] Test the user can't create `camelCase` api from the front-end
  - [x] Test delete API
- **Users Permissions**
  - [x] Test login
  - [x] Test register
- **Content Manager**
  - [x] Test build
  - [x] Test errors  when navigating between all views
  - [x] Test filters
  - [x] Test default sort
  - [x] Test build
  - [x] Test create data with relations
  - [x] Test update data with relation
  - [x] Test validations
- **Monorepo**
  - [x] Add cypress tests in the repository
  - [x] Launch test programmatically
  - [x] Fix back-end current tests **cc/** @lauriejim

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+Cypress.Commands.add('checkModalOpening', () => {
+  return cy.get('.modal').invoke('show');
+})
+
+Cypress.Commands.add('deleteUser', (id, jwt) => {
+  cy.request({
+    url: `${backendUrl}/users/${id}`,
+    method: 'DELETE',
+    headers: {
+      Authorization: `Bearer ${jwt}`
```
Comment (Aurelsicoko) [inline cypress/support/commands.js]: Are you sure there is no way to only pass the Bearer token only once? Then, all the future requests will be automatically authenticated?

File: cypress/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+});
+
+Cypress.Commands.add('login', () => {
+  cy.createUser()
+  return cy.request({
+    url: `${backendUrl}/auth/local`,
+    method: 'POST',
+    body: {
+      identifier: 'soup',
+      password: 'coucou123',
```
Comment (Aurelsicoko) [inline cypress/support/commands.js]: Can you try to put more real and secure identifiers?

File: package.json [hunk 3]
```diff
@@ -1,7 +1,9 @@
 {
   "private": true,
   "version": "3.0.0-alpha.14.5",
-  "dependencies": {},
+  "dependencies": {
+    "cypress": "^3.1.2"
```
Comment (Aurelsicoko) [inline package.json]: It should be a `devDependencies`

File: packages/strapi-plugin-users-permissions/cypress/integration/login_specs.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+const frontLoadingDelay = Cypress.config('frontLoadingDelay');
+const userData = {
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-users-permissions/cypress/integration/login_specs.js]: Can you change these identifiers?

File: packages/strapi-plugin-users-permissions/cypress/integration/register_specs.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+const frontLoadingDelay = Cypress.config('frontLoadingDelay');
+const registerData = {
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-users-permissions/cypress/integration/register_specs.js]: Same 🙈

File: cypress/support/commands.js (hunk 1 above)
Comment (soupette) [reply cypress/support/commands.js]: I'm creating new token every new test

File: packages/strapi-plugin-users-permissions/cypress/integration/login_specs.js (hunk 4 above)
Comment (soupette) [reply packages/strapi-plugin-users-permissions/cypress/integration/login_specs.js]: It's weird it has been changed here https://github.com/strapi/strapi/blob/add-cypress/packages/strapi-plugin-users-permissions/cypress/integration/login_specs.js

File: packages/strapi-plugin-users-permissions/cypress/integration/register_specs.js (hunk 5 above)
Comment (soupette) [reply packages/strapi-plugin-users-permissions/cypress/integration/register_specs.js]: Same https://github.com/strapi/strapi/blob/add-cypress/packages/strapi-plugin-users-permissions/cypress/integration/register_specs.js

File: package.json (hunk 3 above)
Comment (soupette) [reply package.json]: https://github.com/strapi/strapi/blob/d60b5c5b34d4aa72c43083460ad89c927485e487/package.json#L9-L11

**Comment by Aurelsicoko** [comment]
The code LGTM (I didn't test it though) 👍

**Comment by lauriejim** [comment]
Cypress global install is required to launch test @soupette ?

screen shot 2018-12-10 at 16 45 59

**Pull request merged**